homeassistant/components/huum/climate.py

```diff
@@ -0,0 +1,133 @@
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Huum sauna with config flow."""
+    huum_handler = hass.data.setdefault(DOMAIN, {})[entry.entry_id]
+
+    async_add_entities([HuumDevice(huum_handler, entry.entry_id)], True)
+
+
+class HuumDevice(ClimateEntity):
+    """Representation of a heater."""
+
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_max_temp = 110
+    _attr_min_temp = 40
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    _target_temperature = 40
+    _status: HuumStatusResponse | None = None
+
+    def __init__(self, huum_handler: Huum, unique_id: str) -> None:
+        """Initialize the heater."""
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, unique_id)},
+            name="Huum sauna",
+            manufacturer="Huum",
+        )
+
+        self._huum_handler = huum_handler
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool mode."""
+        if self._status and self._status.status == SaunaStatus.ONLINE_HEATING:
+            return HVACMode.HEAT
+        return HVACMode.OFF
+
+    @property
+    def icon(self) -> str:
+        """Return nice icon for heater."""
+        if self.hvac_mode == HVACMode.HEAT:
+            return "mdi:radiator"
+        return "mdi:radiator-off"
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Return the current temperature."""
+        if (status := self._status) is not None:
+            return status.temperature
+        return None
+
+    @property
+    def target_temperature(self) -> int | None:
+        """Return the temperature we try to reach."""
+        if (status := self._status) is not None:
+            return status.target_temperature
+        return None
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set hvac mode."""
+        if hvac_mode == HVACMode.HEAT:
+            temperature = (
+                self.target_temperature
+                if self.target_temperature
+                else int(self.min_temp)
+            )
```

Comment:
So if I understand correctly, target_temperature is now only set when the device is not idle. But since it's not idle, it's heating, and when its already heating, you dont have a reason to set the hvac mode to heat as it already is.

Furthermore, please avoid using multi line inline if statements if they span more than 1 line.

I will discuss this with someone else to make sure this is a right choice

Reply:
The only way to set the target temperature through the API currently is by turning it on, that is unfortunately true. This is not a choice by me, but by the API Huum has provided me. I hope that I get an OK from Huum to start using the same API as they are using in their app soon enough, as that _can_ set the temperature separately, but as of right now that is not possible.

For the code style here, do you have a code style guide somewhere that I could follow, for me multi-line if-else statement like these are commonplace. I would rather focus on the actual logic of the code than the style of it, so it would be greatly helpful if I could check some place for how you want code formatted. With that said, I'll change this to a "normal" if-else statement.

Reply:
But there is actually a bug in there as well, so I will have to fix that. The target temperature that the user has set is not actually used, and it jumps back to the one that we got from the API. I did not use the`self._target_temperature` that I had already created but has mistakenly not used 😄

Reply:
apart from the docs I don't really know if there is a style guide. I am also still learning the little quirks of the style we use

Reply:
I now added one more commit that fixes the settings of the current temperature. Due to the amount of refactoring done, I forgot to change the logic accordingly, sorry. So now the target temperature works and it can be set properly in the UI.

The issue previously was that if the sauna was off and the target temp was set in the UI and then the sauna was turned to heat, then the target temp from the API was used for turning it on and not the temp set by the user in the UI. That should now be fixed. It also simplified some of the code so that was nice :) I would really hope that we get access to the proper APIs, as that would simplify this further and would not require us to do some internal setting of temperature the way it is now done.

Reply:
I guess it works something like this now (note that the code is the real thing to check, this is just a flow chart)

```mermaid
flowchart LR
    load[Load integration] --> status[Get status]
    onUpdate[On update] --> status
    userSet[User sets temperature in UI] --> setTempUser[Set `_target_temperature` from UI]
    setTempUser --> isOn{Is sauna on?}
    isOn -- Yes --> callTemp
    status --> hasTempOrOn{Do we have \n`_target_temperature` \nor is the sauna on?}
    hasTempOrOn -- Yes --> setTempStatus[Set `_target_temperature` from status]
    callTemp[Get `target_temperature`] --> hasTargetTemp{Is `_target_temperature` set?}
    hasTargetTemp -- No --> returnMin[Return min temp]
    hasTargetTemp -- Yes --> returnTemp[Return `_target_temperature`]
    returnMin --> turnOn[Turn on sauna with temperature]
    returnTemp --> turnOn[Turn on sauna with temperature]
```

(PS. made it left-to-right to not take up a ton of space in the PR discussion, top-down is slightly more readable)

Reply:
(what, I did not even know github had support for this. What did you use to create this?)

Reply:
It is [Mermaid](https://mermaid.js.org/) :) It has been supported [for a while](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/), it is also supported on GitLab